### PR TITLE
Updating Sample application to add Carthage and a Makefile for convenience. 

### PR DESCRIPTION
**Pull request by jdizoglio**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

File: BoxContentSDKSampleApp/Makefile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+CARTHAGE := $(shell command -v carthage 2> /dev/null)
+
+.PHONY: help
```
Comment (jlawton) [inline BoxContentSDKSampleApp/Makefile]: All of these rules are technically .PHONY. Maybe we should mark that, too, to be clean.

File: BoxContentSDKSampleApp/Makefile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```
Comment (jlawton) [inline BoxContentSDKSampleApp/Makefile]: Neat. That's a cool approach.

File: BoxContentSDKSampleApp/Makefile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+bootstrap: check_carthage ## Build the reference app dependency framework(s) to cartfile.resolved
+	# Make sure submodules handled by Carthage are up to date (as saved in Cartfile.resolved)
+	$(CARTHAGE) bootstrap --project-directory . --platform iOS
```
Comment (jlawton) [inline BoxContentSDKSampleApp/Makefile]: No real need to have `--project-directory .` in any of these commands.

File: BoxContentSDKSampleApp/Cartfile [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Box SDKs
+git "git@github.com:box/box-ios-sdk.git" ~> 1.0
```
Comment (jlawton) [inline BoxContentSDKSampleApp/Cartfile]: We could also use `github "box/box-ios-sdk" ~> 1.0`. I'm not sure if that affects whether Carthage tries to use binaries from Github releases, but maybe. Carthage would then use https by default, but I think that's fine for public repos.

File: BoxContentSDKSampleApp/Makefile (hunk 1 above)
Comment (jdizoglio) [reply BoxContentSDKSampleApp/Makefile]: done

File: BoxContentSDKSampleApp/Makefile (hunk 3 above)
Comment (jdizoglio) [reply BoxContentSDKSampleApp/Makefile]: done

File: BoxContentSDKSampleApp/Cartfile (hunk 4 above)
Comment (jdizoglio) [reply BoxContentSDKSampleApp/Cartfile]: done

**Review by jlawton — APPROVED** [review]

File: BoxContentSDKSampleApp/Cartfile [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# Box SDKs
+git "box/box-ios-sdk" ~> 1.0
```
Comment (jlawton) [inline BoxContentSDKSampleApp/Cartfile:2]: `github`

**Pull request merged**